src/parsers/manifest/metaplaylist/index.ts

```diff
@@ -0,0 +1,202 @@
+    .reduce((acc, val) =>
+      Math.min(acc || DEFAULT_LIVE_GAP, val || DEFAULT_LIVE_GAP), DEFAULT_LIVE_GAP
+    )) || DEFAULT_LIVE_GAP;
+
+  const suggestedPresentationDelay = // TODO root level in MPL?
+    (contents.map(content => content.manifest.suggestedPresentationDelay)
+      .reduce((acc, val) => Math.min(acc || 10, val || 10), 10)) || 10;
+
+  const maxSegmentDuration = // TODO root level in MPL?
+    contents.map(content => content.manifest.maxSegmentDuration)
+      .reduce((acc, val) => Math.min((acc || 0), (val || 0)), 0);
+
+  const minBufferTime = // TODO root level in MPL?
+    contents.map(content => content.manifest.minBufferTime)
+      .reduce((acc, val) => Math.min((acc || 0), (val || 0)), 0);
+
+  // Build new period array
+  const newPeriods: IParsedPeriod[] = [];
+  const contentEnding = contents[contents.length - 1].endTime;
+
+  for (let j = 0; j < parsedPeriods.length; j++) {
+    const baseManifest = new Manifest(
+      contents[j].manifest as IManifestArguments,
+      new Subject(), // TODO Warn about problems?
+      {}
+    );
+
+    const parsedPeriod = parsedPeriods[j];
+    parsedPeriod.start = contents[j].startTime;
+    parsedPeriod.end =
+      contents[j].endTime || (parsedPeriod.start + (parsedPeriod.duration || 0));
+    parsedPeriod.duration = parsedPeriod.end - parsedPeriod.start;
+
+    const textTracks = contents[j].textTracks;
+    if (textTracks && textTracks.length > 0) {
+      textTracks.forEach((track) => {
+        const textAdaptation = {
+          id: "gen-text-ada-" + generateNewId(),
+          representations: [{
+            mimeType: track.mimeType,
+            bitrate: 0,
+            index: new StaticRepresentationIndex({ media: track.url }),
+            id: "gen-text-rep-" + generateNewId(),
+          }],
+          type: "text",
+          normalizedLanguage: track.language,
+        };
+        if (!parsedPeriod.adaptations.text) {
+          parsedPeriod.adaptations.text = [];
+        }
+        parsedPeriod.adaptations.text.push(textAdaptation);
+      });
+    }
+
+    const adaptations = parsedPeriod.adaptations;
+    SUPPORTED_ADAPTATIONS_TYPE.forEach((adaptationType) => {
+      const adaptationsByType = adaptations[adaptationType];
+      if (adaptationsByType) {
+        adaptationsByType.forEach((adaptation) => {
+          const representations = adaptation.representations;
+          representations.forEach((representation) => {
+            const index = representation.index;
+
+            // XXX TODO What?
+            // Store base contents info
+
+            const basePeriod =
+              arrayFind(baseManifest.periods, p => p.id === parsedPeriod.id);
+            if (basePeriod == null) {
+              throw new Error();
+            }
+
+            const whatwhatwhat = ["audio", "video", "image", "text"]
+              .reduce((acc: Adaptation[], type) => {
+                const _adaptation = (baseManifest.periods[0] && baseManifest.periods[0])
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Correct one of the identical sub-expressions on both sides of operator "&&" ([typescript:S1764](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS1764))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=367&issues=AWd1fPma8YaV-M4wPwBi&open=AWd1fPma8YaV-M4wPwBi)

[ISSUE_KEY]: AWd1fPma8YaV-M4wPwBi